### PR TITLE
Feature sb 45 my profile complete form

### PR DESCRIPTION
**Pull request by VladManukian**
### **_Changes:_**
1) Added update endpoint
2) Added registration wrapper
3) Reworked public/private routes provider
4) Added common UseFormHook
5) Eslint rules reworked ("sonarjs/no-duplicate-string": "off")
6) Delete header/fotter wrapper. Added this element into common wrapper
### **Screen:**
form

### CODE REVIEW

File: src/components/Footer/Footer.tsx [hunk 1]
```diff
@@ -34,8 +35,19 @@ export const Footer: React.FC = () => {
 		? socialMediaDataLogIn
 		: socialMediaDataLogOut;
 
+	const NoHeaderRoutes = [
```
Comment (lyudad) [inline src/components/Footer/Footer.tsx]: why name start from upper case

Comment (lyudad) [reply src/components/Footer/Footer.tsx]: move above the component, we dont need reinitialisation with each rerender

File: src/components/Header/Header.tsx [hunk 2]
```diff
@@ -26,12 +27,19 @@ import { type RootState } from "~/redux/store.ts";
...
 	const { t } = useTranslation();
+	const location = useLocation();
 	const isLoggedIn = Boolean(
 		useAppSelector((state: RootState) => state.auth.accessToken),
 	);
+	const NoHeaderRoutes = [
+		AppRoute.ROOT,
+		AppRoute.SIGN_UP,
+		AppRoute.SIGN_IN,
+	] as string[];
```
Comment (lyudad) [inline src/components/Header/Header.tsx]: you use the same constant 2ce, why not move it to general constant file?

File: src/components/image-preview/image-preview.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+const ImagePreview: React.FC<{ file: File | null }> = ({ file }) => {
+	return (
+		<Box
+			alignItems="center"
+			display="flex"
+			flexDirection="column"
+			gap={1}
+			sx={{
+				border: "1px solid #E3EEE2",
```
Comment (lyudad) [inline src/components/image-preview/image-preview.tsx]: 1.3.2

File: src/components/image-preview/image-preview.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+				height: "120px",
+				overflow: "hidden",
+				width: "120px",
+			}}
+		>
+			{file ? (
+				<img
+					alt="Uploaded"
+					src={URL.createObjectURL(file)}
+					style={{ height: "100%", objectFit: "cover", width: "100%" }}
```
Comment (lyudad) [inline src/components/image-preview/image-preview.tsx]: 1.3.4

File: src/components/Footer/Footer.tsx (hunk 1 above)
Comment (VladManukian) [reply src/components/Footer/Footer.tsx]: Deleted and now i use another logic.

File: src/components/radio-button/index.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	backgroundImage:
+		"linear-gradient(180deg,hsla(0,0%,100%,.05),hsla(0,0%,100%,0))",
+	borderRadius: "50%",
+	boxShadow: `0 0 0 1px ${theme.palette.darkGrey}`,
+	height: 16,
+	width: 16,
+}));
+
+const CustomCheckedIcon = styled(CustomIcon)(({ theme }) => ({
+	"&::before": {
```
Comment (ValentinaTenus) [inline src/components/radio-button/index.tsx]: Shouldn't it be in separate file styles.ts?

File: src/pages/profile-board/components/address-form.tsx [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+	);
+
+	return (
+		<StyledFormContainer>
+			<Box width="30%">
+				<Typography
+					color="primary"
+					marginBottom="8px"
+					sx={{ fontSize: theme.typography.playfairDisplayBold }}
+					variant="h6"
```
Comment (ValentinaTenus) [inline src/pages/profile-board/components/address-form.tsx]: Does it work for you? you can use just variant="playfairDisplayBold" if it has size, fontWeight which you need. or override some of them with sx

File: src/pages/profile-board/helpers/expire-date-change.helper.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+	const secondDigit = formattedValue.charAt(ONE);
+
+	if (firstDigit !== "0" && firstDigit !== "1") {
+		formattedValue = "";
+	} else if (firstDigit === "0" && secondDigit && !/[1-9]/.test(secondDigit)) {
+		formattedValue = "0" + secondDigit;
+	} else if (
+		firstDigit === "1" &&
+		secondDigit &&
+		(parseInt(secondDigit, 10) > TWO || isNaN(parseInt(secondDigit, 10)))
```
Comment (ValentinaTenus) [inline src/pages/profile-board/helpers/expire-date-change.helper.ts:33]: magic numbers

**Review by ValentinaTenus — APPROVED** [review]

**Review by likeri29 — APPROVED** [review]

File: src/components/auth-route/index.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+		if (user.isVerified && user.profile.isRegistrationCompleted)
+			return <Navigate to={AppRoute.HOME} />;
+
+		if (!user?.isVerified) return <Navigate to={AppRoute.VERIFY_EMAIL} />;
+
+		if (!user?.profile.role) return <Navigate to={AppRoute.ROLE} />;
+
+		if (!user?.profile.isRegistrationCompleted)
+			return <Navigate to={AppRoute.GENERAL_INFORMATION} />;
+	}
```
Comment (jaroslawtt) [inline src/components/auth-route/index.tsx]: optional chaining is redudant

**Review by jaroslawtt — APPROVED** [review]

**Pull request merged**